code/_onclick/xeno.dm

```diff
@@ -1,100 +1,76 @@
-			return 1
-
-	return ..()
-
-/mob/living/carbon/Xenomorph/Boiler/click(var/atom/A, var/list/mods)
-	if(!istype(A,/obj/screen) && is_bombarding)
-		if(isturf(A))
-			bomb_turf(A)
-		else
-			bomb_turf(get_turf(A))
-		if(client)
-			client.mouse_pointer_icon = initial(client.mouse_pointer_icon)
-		return TRUE
-	if(mods["middle"] && !mods["shift"])
-		if (selected_ability && middle_mouse_toggle)
-			selected_ability.use_ability(A)
-			return TRUE
-	if(mods["shift"])
-		if (selected_ability && !middle_mouse_toggle)
-			selected_ability.use_ability(A)
-			return TRUE
-	return ..()
-
-/mob/living/carbon/Xenomorph/Crusher/click(var/atom/A, var/list/mods)
-	if(!istype(A, /obj/screen))
-		if(is_charging)
-			stop_momentum(charge_dir)
-
-	if(mods["middle"] && !mods["shift"])
-		if(selected_ability && middle_mouse_toggle)
-			selected_ability.use_ability(A)
-			return 1
-
-	if(mods["shift"])
-		if(selected_ability && !middle_mouse_toggle)
-			selected_ability.use_ability(A)
-			return 1
-
-	return ..()
-
-/mob/living/carbon/Xenomorph/Larva/UnarmedAttack(var/atom/A, var/list/mods)
-
-	if(lying) //No attacks while laying down
-		return 0
+
+/mob/living/carbon/Xenomorph/Larva/UnarmedAttack(atom/A, proximity_flag)
+	if(lying)
+		return FALSE
 
 	A.attack_larva(src)
-	next_move = world.time + (10 + attack_delay) //Adds some lag to the 'attack'
+
 
 //Larva attack, will default to attack_alien behaviour unless overriden
 /atom/proc/attack_larva(mob/living/carbon/Xenomorph/Larva/user)
 	return attack_alien(user)
 
-/mob/living/carbon/Xenomorph/Queen/click(var/atom/A, var/list/mods)
-
-	if(mods["ctrl"] && mods["middle"])
-		if(ovipositor)
-			if(isxeno(A) && A != src)
-				var/mob/living/carbon/Xenomorph/X = A
-				if(X.stat != DEAD)
-					set_queen_overwatch(A)
-					return 1
-
-	if(mods["middle"] && !mods["shift"])
-		if (selected_ability && middle_mouse_toggle)
-			selected_ability.use_ability(A)
-			return 1
-
-	if(mods["shift"])
-		if(selected_ability && !middle_mouse_toggle)
-			selected_ability.use_ability(A)
-			return 1
+
+/mob/living/carbon/Xenomorph/Queen/CtrlMiddleClickOn(atom/A)
+	. = ..()
+	if(!ovipositor)
+		return
+	if(!isxeno(A) || A == src)
+		return
+	var/mob/living/carbon/Xenomorph/X = A
+	if(X.stat == DEAD)
+		return
+	set_queen_overwatch(A)
+
+
+/mob/living/carbon/Xenomorph/MiddleClickOn(atom/A)
+	. = ..()
+	if(!middle_mouse_toggle || !selected_ability)
+		return
+	selected_ability.use_ability(A)
+
+
+/mob/living/carbon/Xenomorph/ShiftClickOn(atom/A)
+	. = ..()
+	if(!selected_ability || middle_mouse_toggle)
+		return
+	selected_ability.use_ability(A)
+
+
+
+/mob/living/carbon/Xenomorph/Boiler/ClickOn(atom/A, params)
+	. = ..()
+	if(istype(A, /obj/screen) || !is_bombarding)
+		return
+	if(isturf(A))
+		bomb_turf(A)
+	else
+		bomb_turf(get_turf(A))
+	if(client)
+		client.mouse_pointer_icon = initial(client.mouse_pointer_icon)
+
+
+/mob/living/carbon/Xenomorph/Crusher/ClickOn(atom/A, params)
+	. = ..()
+	if(istype(A, /obj/screen))
```

Comment:
Same here, `/obj/screen/Click()` should prevent the click from reaching here, no?

Reply:
Same